internal/clients/api_client.go

```diff
@@ -93,6 +97,35 @@ func NewAcceptanceTestingClient() (*ApiClient, error) {
 
 const esConnectionKey string = "elasticsearch_connection"
 
+type ElasticSearchConnection struct {
```

Comment:
IMO it would be cleaner to have all functions which build a new client parse the relevant config into this struct and then call a single function which builds an `elasticsearch.Config{}` instance from this struct.

It would mean consolidating all the CA, TLS and debugging code. Given the different diagnostic types we'd have to return a plain error but I think that's ok. 

WDYT?